code/game/objects/items/explosives/mine.dm

```diff
@@ -72,47 +112,65 @@
 
 	user.visible_message(SPAN_NOTICE("[user] starts deploying [src]."), \
 		SPAN_NOTICE("You start deploying [src]."))
-	if(!do_after(user, 40, INTERRUPT_NO_NEEDHAND, BUSY_ICON_HOSTILE))
+	playsound(loc, 'sound/machines/click.ogg', 25, 1)
+	if(!do_after(user, deploy_time, INTERRUPT_NO_NEEDHAND, BUSY_ICON_HOSTILE))
 		user.visible_message(SPAN_NOTICE("[user] stops deploying [src]."), \
-			SPAN_NOTICE("You stop deploying \the [src]."))
+			SPAN_NOTICE("You stop deploying [src]."))
 		return
 
 	if(active)
 		return
 
 	if(check_for_obstacles(user))
 		return
-
-	user.visible_message(SPAN_NOTICE("[user] finishes deploying [src]."), \
+	if(needs_digging)
+		playsound(loc, 'sound/weapons/flipblade.ogg', 25, 1)
+		user.visible_message(SPAN_NOTICE("[user] pulls the pin on [src]."), \
+			SPAN_NOTICE("You pull the activation pin and prepare it to be buried."))
+		user.drop_inv_item_on_ground(src)
+		anchored = TRUE
+		return
+	else
+		user.visible_message(SPAN_NOTICE("[user] finishes deploying [src]."), \
 		SPAN_NOTICE("You finish deploying [src]."))
-
+		if(user)
+			user.drop_inv_item_on_ground(src)
+			setDir(user ? user.dir : dir)
 	deploy_mine(user)
 
 /obj/item/explosive/mine/proc/deploy_mine(mob/user)
 	if(!hard_iff_lock && user)
 		iff_signal = user.faction
-
 	cause_data = create_cause_data(initial(name), user)
 	anchored = TRUE
-	playsound(loc, 'sound/weapons/mine_armed.ogg', 25, 1)
-	if(user)
-		user.drop_inv_item_on_ground(src)
-	setDir(user ? user.dir : dir) //The direction it is planted in is the direction the user faces at that time
+	playsound(loc, 'sound/weapons/mine_armed.ogg', 25, 1) //The direction it is planted in is the direction the user faces at that time
 	activate_sensors()
 	update_icon()
 
 
 //Disarming
-/obj/item/explosive/mine/attackby(obj/item/W, mob/user)
-	if(HAS_TRAIT(W, TRAIT_TOOL_MULTITOOL))
+/obj/item/explosive/mine/attackby(obj/item/thing, mob/user)
+	if(needs_digging && HAS_TRAIT(thing,  TRAIT_TOOL_SHOVEL) && !active && anchored && !buried)
+		user.visible_message(SPAN_NOTICE("[user] starts burying [src]."), \
+			SPAN_NOTICE("You start burying [src]."))
+		playsound(user.loc, 'sound/effects/thud.ogg', 40, 1, 6)
+		if(!do_after(user, 3 SECONDS, INTERRUPT_NO_NEEDHAND, BUSY_ICON_BUILD))
+			user.visible_message(SPAN_WARNING("[user] stops burying [src]."), \
+			SPAN_WARNING("You stop burying [src]."))
+			return
+		user.visible_message(SPAN_NOTICE("[user] finished burying [src]."), \
```

Comment:
check again for anchored/active/buried/needs_digging (or whichever are appropriate) post do_after here

Reply:
What?

Reply:
You do checks to verify the mine is in a valid state to be burried.

But if for example several persons do it at same time, the state of the mine could be incorrect after the period waited in do_after.

So you need to re-check the mine is in a proper state after the wait to make sure the action is still valid

Reply:
Similar to what's done just below for disarming with `if(!active)//someone beat us to it`